src/Console/DeployCommand.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Laravel\Nightwatch\Console;
+
+use Carbon\CarbonImmutable;
+use Illuminate\Console\Command;
+use Illuminate\Http\Client\RequestException;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Str;
+use SensitiveParameter;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Throwable;
+
+use function config;
+
+/**
+ * @internal
+ */
+#[AsCommand(name: 'nightwatch:deploy', description: 'Notify Nightwatch of a deployment.')]
+final class DeployCommand extends Command
+{
+    /**
+     * @var string
+     */
+    protected $signature = 'nightwatch:deploy';
+
+    /**
+     * @var string
+     */
+    protected $description = 'Notify Nightwatch of a deployment.';
+
+    /**
+     * @var bool
+     */
+    protected $hidden = true;
+
+    public function __construct(
+        #[SensitiveParameter] private ?string $token,
+    ) {
+        parent::__construct();
+    }
+
+    public function handle(): int
+    {
+        if (! $this->token) {
+            $this->components->error('Please configure the [NIGHTWATCH_TOKEN] environment variable.');
+
+            return 1;
+        }
+
+        $tag = config('nightwatch.deployment') ?? '';
+
+        $baseUrl = ! empty($_SERVER['NIGHTWATCH_BASE_URL']) ? $_SERVER['NIGHTWATCH_BASE_URL'] : 'https://nightwatch.laravel.com';
+
+        try {
+            Http::connectTimeout(5)
+                ->timeout(10)
+                ->withHeaders([
+                    'Authorization' => "Bearer {$this->token}",
+                    'Accept' => 'application/json',
+                ])
+                ->post("{$baseUrl}/api/deployments", [
+                    'v' => 1,
+                    'timestamp' => CarbonImmutable::now()->timestamp,
+                    'version' => $tag,
+                ])
+                ->throw();
+
+            $this->components->info('Deployment sent to Nightwatch successfully.');
+
+            return 0;
+        } catch (RequestException $e) {
+            $message = Str::limit($e->response->json('message') ?? "[{$e->getCode()}] {$e->response->body()}", 1000, '[...]'); // @phpstan-ignore argument.type
+
+            $this->components->error("Deployment could not be sent to Nightwatch: {$message}");
+
+            return 1;
+        } catch (Throwable $e) {
+            $this->components->error("Deployment could not be sent to Nightwatch: {$e->getMessage()}");
+
+            return 1;
```

Comment:
I have mixed feelings on returning non-zero here.

If someone has their deploys configured to stop on an error, then this could block their deploy (unless they run something like `php artisan nightwatch:deploy || true`)

On the other hand, it also feels weird to return 0 when there was an error.

I'm leaning towards returning 0 by default, but perhaps adding a CLI option to enable returning non-zero on error.

Reply:
Great point and I agree.

Nightwatch should never be the cause of an error, only help you debug it. We have gone to great lengths to make this a fact throughout the package and it makes sense that we continue that here.

We should call `report` or `Nightwatch::report` and hopefully the error would surface in Nightwatch itself.

Maybe we don't add a CLI flag until someone asks for it?